### PR TITLE
Readme: statement of need

### PR DESCRIPTION
**Pull request by elboyran**
Addresses #144 and #138 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,19 +42,19 @@ affiliations:
...
+
+DIANNA software is addressing needs of both (X)AI reseachers and mostly the various domains scientists who are using or will use AI models for their research without being experts in (X)AI. DIANNA is future-proof: the only XAI library supporting the [Open Neural Network Exchange (ONNX)](https://onnx.ai/) format. 
+
+* Provides an easy-to-use interface for non (X)AI experts
+* Implements well-known XAI methods (LIME, RISE and Kernal SHAP) chosen by systematic and objective evaluation criteria
+* Supports the de-facto standart of neural network models - ONNX.
+* Includes clear instructions for export/conversions from Tensorflow, Pytorch, Keras and skikit-learn to ONNX.
+* Supports both images and text data modalities. Time series, tabular data and even embeddings support is planned.
+* Comes with simple intuitive image and text benchmarks 
+* Easily extendable to other XAI methods
```
Comment (egpbos) [inline README.md:55]: Oops, I should have looked at this before doing #178. I think we should merge this with what I wrote there, otherwise we'll be repeating ourselves.

Comment (egpbos) [reply README.md:55]: I think btw that if we shorten and merge these two sections, we can leave the "Why DIANNA?" header out.

**Review by egpbos — COMMENTED** [review]
Have to stop now, I will look at CONTEXT.md later.

File: CONTEXT.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+* **A single DNN format/framework/architecture is supported**
+
+Lucid supports only [Tensoflow](https://www.tensorflow.org/), Captum is the library for [PyTorch](https://pytorch.org/) users, [iNNvstigate](https://github.com/albermax/innvestigate) has been used by Keras users. Pytorch-grad-cam supports a single method for a single format and Pytorch-cnn-visualizaiton even limits the choice to a single DNN type- CNN. While this is not an issue for the current most popular framework communities, not all are supported by mature libraries (e.g. ) and most importantly are not "future-proof", e.g. [Caffe](https://caffe.berkeleyvision.org/) has been the most polular framework by the computer vision comminuty in 2018, while now is the era of Tensorflow, Pytorch and Keras.  
+
+* **The choice of supported XAI methods seems random**
+
+ELI5, for example, supports multiple frameworks/formats and XAI methods, but is not clear how the selection of these methods has been made. Futhermore the library is not maintained since 2020 and hence lacks any methods proposed since. Again, which are the best methods to use?
+
```
Comment (cwmeijer) [inline CONTEXT.md:29]: To be honest, I don't think anyone would understand our choices either. Of course we started out, kind of, systematic but that was already arguably subjective. After all the technical problems with onnx-tf and onnx-torch we also ended up with set that is hard to identify by any user as well thought out. I'm ok to keep this argument, but I'm not really sure if it holds.

File: CONTEXT.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+All of the above issues have a common reason: _the libraries have been developed by (X)AI reseachers for the (X)AI reseachrs_.
+ 
+Even for the purpose of advancing (X)AI research, systematic study of the properties of the relevance "heatmaps" is lacking and currently in the XAI literature often the human interpretation is intertwined with the one of the exlainer. Suitable datasets for such a study are lacking: the popular and simplest dataset used in publicaitons,  MNIST, is too complex for such a prurpose with 10 classes and no structural content variation. The XAI community does not publish work on simple scientific datasets, which would make the technology understanable and trustable by non-expert scientists on intuitive level. Another important show-stopper for reproducible AI-enabled science is the pletora of current formats and the speed in which they become obsolete. Some libaries are also becoming obsolete for the lack of support of the research projects which have sponsored their creation (e.g. iNNvestigate).
```
Comment (cwmeijer) [inline CONTEXT.md:35]:
```suggestion
Even for the purpose of advancing (X)AI research, systematic study of the properties of the relevance "heatmaps" is lacking and currently in the XAI literature often the human interpretation is intertwined with the one of the exlainer. Suitable datasets for such a study are lacking: the popular and simplest dataset used in publicaitons,  MNIST, is too complex for such a prurpose with 10 classes and no structural content variation. The XAI community does not publish work on simple scientific datasets, which would make the technology understandable and trustworthy by non-expert scientists on intuitive level. Another important show-stopper for reproducible AI-enabled science is the plethora of current formats and the speed in which they become obsolete. Some libraries are also becoming obsolete for the lack of support of the research projects which have sponsored their creation (e.g. iNNvestigate).
```

File: CONTEXT.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+* **The choice of supported XAI methods seems random**
+
+ELI5, for example, supports multiple frameworks/formats and XAI methods, but is not clear how the selection of these methods has been made. Futhermore the library is not maintained since 2020 and hence lacks any methods proposed since. Again, which are the best methods to use?
+
+* **Requires (X)AI expertise**
+For example the DeepVis Toolbox requires DNN knowledge and is used only by AI experts mostly within the computer vision community (which is large and can explain the number of stars). 
+
+All of the above issues have a common reason: _the libraries have been developed by (X)AI reseachers for the (X)AI reseachrs_.
```
Comment (cwmeijer) [inline CONTEXT.md:33]:
```suggestion
All of the above issues have a common reason: _the libraries have been developed by (X)AI researchers for the (X)AI researchers_.
```

File: CONTEXT.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ 
+*	**Systematically chosen generic explainers**
+
+DIANNA includes few XAI approaches which have been evaluated using [systematically defined criteria](https://arxiv.org/ftp/arxiv/papers/1912/1912.05100.pdf). Using a relevant subset of these criteria, we have included complementary state-of-the-art XAI methods: [LIME](https://www.kdd.org/kdd2016/papers/files/rfp0573-ribeiroA.pdf), [RISE](http://bmvc2018.org/contents/papers/1064.pdf) and [KernelSHAP](https://proceedings.neurips.cc/paper/2017/file/8a20a8621978632d76c43dfd28b67767-Paper.pdf). These methods are model architecture agnostic.
+
+*	**Standart DNN format**
```
Comment (cwmeijer) [inline CONTEXT.md:48]:
```suggestion
*	**Standard DNN format**
```

File: CONTEXT.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+*	**Simple datasets**
+
+To demonstrate the usefulness and enable studying the properties of the "heatmaps" on an intuitive level, DIANNA comes with two new datasets: [Simple Geometric](https://doi.org/10.5281/zenodo.5012824)<img width="20" alt="SimpleGeometric Logo" src="https://user-images.githubusercontent.com/3244249/151817429-80f38846-8a4b-4471-a4c9-df7d19b668e5.png">
+ and [Simple Scientific (LeafSnap30)](https://doi.org/10.5281/zenodo.5061352)<img width="20" alt="LeafSnap30 Logo" src="https://user-images.githubusercontent.com/3244249/151817480-649ad3b7-2b4b-4aa3-a5d6-ebfaa6aa614a.png"> – subset of [LeafSnap dataset](http://leafsnap.com/dataset/). Tree species classification on the LeafSnap data is a good simple scientific example of a problem tackled with both classical Computer Vision in the past and the superior DL method. Enhanced with explaianble tool, the DL approach is a clear winnner. DIANNA also uses 2 well-established bencharks - a (modified to binary) MNIST and the [Stanford Sentimet Treebank](https://nlp.stanford.edu/sentiment/index.html).
```
Comment (cwmeijer) [inline CONTEXT.md:42]:
```suggestion
 and [Simple Scientific (LeafSnap30)](https://doi.org/10.5281/zenodo.5061352)<img width="20" alt="LeafSnap30 Logo" src="https://user-images.githubusercontent.com/3244249/151817480-649ad3b7-2b4b-4aa3-a5d6-ebfaa6aa614a.png"> – subset of [LeafSnap dataset](http://leafsnap.com/dataset/). Tree species classification on the LeafSnap data is a good simple scientific example of a problem tackled with both classical Computer Vision in the past and the superior DL method. Enhanced with explainable tool, the DL approach is a clear winner. DIANNA also uses 2 well-established benchmarks - a (modified to binary) MNIST and the [Stanford Sentiment Treebank](https://nlp.stanford.edu/sentiment/index.html).
```

**Review by cwmeijer — APPROVED** [review]
Looks good!
I have to say, I don't have the overview of the readme its roadmap so I reviewed the text more generically.

**Comment by elboyran** [comment]
The stanford site is down today, will be up at some point. Merging.
